# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: third_party/libwebrtc/modules/video_capture/video_capture.h
# Commit: 30340541c9a1
# Full Hash: 30340541c9a17e21f1c8f7e65850afaa5e3355db
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2023-03-06 16:15:33
# Regressor Bug: 1817724
# File Overlap Count: 3
# Description:
#   Bug 1817724 - Crash in case VideoCaptureAVFoundation is destroyed unexpectedly. r=karlt,webrtc-reviewers,ng
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D171470
# ==============================================================================

diff -r 46e97efba193 -r 30340541c9a1 third_party/libwebrtc/modules/video_capture/video_capture.h
--- a/third_party/libwebrtc/modules/video_capture/video_capture.h	Mon Mar 06 06:58:34 2023 +0000
+++ b/third_party/libwebrtc/modules/video_capture/video_capture.h	Mon Mar 06 07:41:28 2023 +0000
@@ -158,8 +158,13 @@
   // Mozilla: TrackingId setter for use in profiler markers.
   virtual void SetTrackingId(uint32_t aTrackingIdProcId) {}
 
+  // Mozilla: Diagnostic
+  void NotifyReleasing() { mOkToDestroy = true; }
+
  protected:
   ~VideoCaptureModule() override {}
+
+  std::atomic<bool> mOkToDestroy = {false};
 };
 
 }  // namespace webrtc
